src/huggingface_hub/repocard.py

```diff
@@ -265,149 +690,91 @@ def metadata_update(
             especially useful if the repo is updated / committed to concurrently.
     Returns:
         `str`: URL of the commit which updated the card metadata.
+
+    Example:
+        ```python
+        >>> from huggingface_hub import metadata_update
+        >>> metadata = {'model-index': [{'name': 'RoBERTa fine-tuned on ReactionGIF',
+        ...             'results': [{'dataset': {'name': 'ReactionGIF',
+        ...                                      'type': 'julien-c/reactiongif'},
+        ...                           'metrics': [{'name': 'Recall',
+        ...                                        'type': 'recall',
+        ...                                        'value': 0.7762102282047272}],
+        ...                          'task': {'name': 'Text Classification',
+        ...                                   'type': 'text-classification'}}]}]}
+        >>> url = metadata_update("julien-c/reactiongif-roberta", metadata)
```

Comment:
TODO - use internal testing repo instead. This one keeps getting blown up whenever I run doctests 😂